cdc/log.cc

```diff
@@ -938,9 +938,11 @@ cdc::cdc_service::impl::augment_mutation_call(lowres_clock::time_point timeout,
                 auto& m = mutations[idx];
                 auto& s = m.schema();
                 if (should_split(m, *s)) {
-                    for (auto&& mm : split(m, s)) {
-                        mutations.push_back(trans.transform(mm, rs.get()));
-                    }
+                    std::vector<mutation> res;
```

Comment:
Why can't we just push back straight to the mutations?

Reply:
Is it because insertion to mutations can invalidate m?

Reply:
It seems that base_mutation is only used in for_each_change before the call to f. If you store partition key from base_mutation in a variable you should be able to insert mm straight to mutations.